packages/editor-ui/src/components/CredentialEdit/CredentialInputs.vue

```diff
@@ -12,52 +12,51 @@
 			<ParameterInputExpanded
 				v-else
 				:parameter="parameter"
-				:value="credentialData[parameter.name]"
+				:value="getCredentialData(parameter.name)"
 				:documentation-url="documentationUrl"
 				:show-validation-warnings="showValidationWarnings"
-				:label="label"
+				:label="{ size: 'medium' }"
 				event-source="credentials"
 				@update="valueChanged"
 			/>
 		</form>
 	</div>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
-import type { IParameterLabel } from 'n8n-workflow';
+<script setup lang="ts">
+import type {
+	ICredentialDataDecryptedObject,
+	INodeProperties,
+	NodeParameterValueType,
+} from 'n8n-workflow';
 import type { IUpdateInformation } from '@/Interface';
 import ParameterInputExpanded from '../ParameterInputExpanded.vue';
 
-export default defineComponent({
-	name: 'CredentialsInput',
-	components: {
-		ParameterInputExpanded,
-	},
-	props: [
-		'credentialProperties',
-		'credentialData', // ICredentialsDecryptedResponse
-		'documentationUrl',
-		'showValidationWarnings',
-	],
-	data(): { label: IParameterLabel } {
-		return {
-			label: {
-				size: 'medium',
-			},
-		};
-	},
-	methods: {
-		valueChanged(parameterData: IUpdateInformation) {
-			const name = parameterData.name.split('.').pop();
+type Props = {
+	credentialProperties: INodeProperties[];
+	credentialData: ICredentialDataDecryptedObject;
+	documentationUrl: string;
+	showValidationWarnings?: boolean;
+};
 
-			this.$emit('update', {
-				name,
-				value: parameterData.value,
-			});
-		},
-	},
-});
+const props = defineProps<Props>();
+
+const emit = defineEmits<{
+	(event: 'update', value: IUpdateInformation): void;
+}>();
+
+function valueChanged(parameterData: IUpdateInformation) {
+	const name = parameterData.name.split('.').pop() ?? parameterData.name;
+
+	emit('update', {
+		name,
+		value: parameterData.value,
+	});
+}
+
+function getCredentialData(parameterName: string) {
+	return props.credentialData[parameterName] as NodeParameterValueType;
+}
```

Comment:
I'd avoid switching to a get function here since it won't get cached. If you need to add the custom typing you could use a computed prop.